recipe/recipe.go

```diff
@@ -1,30 +1,18 @@
 package recipe
 
-// SourceRecipe contains the json data for a recipe that is used to generate
-// the source code for a recipe.
-type SourceRecipe struct {
-	Type   string                 `json:"type" yaml:"type" validate:"required"`
-	Config map[string]interface{} `json:"config" yaml:"config"`
-}
-
-// SinkRecipe contains the json data for a recipe that is being used for
-// generating the sink code for a recipe.
-type SinkRecipe struct {
-	Name   string                 `json:"name" yaml:"name" validate:"required"`
-	Config map[string]interface{} `json:"config" yaml:"config"`
+// Recipe contains the json data for a recipe
+type Recipe struct {
+	Name       string         `json:"name" yaml:"name" validate:"required"`
+	Source     PluginRecipe   `json:"source" yaml:"source" validate:"required"`
+	Sinks      []PluginRecipe `json:"sinks" yaml:"sinks" validate:"required,min=1"`
+	Processors []PluginRecipe `json:"processors" yaml:"processors"`
+	Node       RecipeNode
 }
 
-// ProcessorRecipe contains the json data for a recipe that is being used for
-// generating the processor code for a recipe.
-type ProcessorRecipe struct {
+// PluginRecipe contains the json data for a recipe that is being used for
+// generating the plugins code for a recipe.
+type PluginRecipe struct {
 	Name   string                 `json:"name" yaml:"name" validate:"required"`
```

Comment:
@scortier  This is a breaking changes, old recipes will fail because `source` is still using `type` in the recipe yaml file.

Should we handle backward comp to not break old recipes? We could support both `type` and `name` and then deprecate `type` later.